Add example with iplotx

iosonofabio opened the pull request:
This PR aims at adding an example to the tree drawing documentation, which shows how to use `cogent3` together with `iplotx` to visualise trees. Draft PR for now to get any feedback from the devs.

I tried to run the doc builder locally and could almost manage by doing the following steps:

```bash
uv sync --extra doc
cd doc
uv run  sphinx-build . ../html/
```

It does run for a long time (until ~72%) and eventually errors out with the message:

```python
TypeError: Can't instantiate abstract class AlignedSeqsData without an implementation for abstract methods 'get_pos_range', 'variable_positions'
```

Assuming this is interesting as an addition to the docs, I'd be happy to ensure it works locally with a little help to overcome this error message.

## Summary by Sourcery

Add a new example to the documentation demonstrating how to use cogent3 with iplotx for tree visualization and update related dependencies accordingly.

New Features:
- Add a tree visualization example using iplotx in the documentation

Enhancements:
- Bump Python requirement to >=3.11
- Pin cogent3-h5seqs dependency to >=0.6.1
- Add iplotx>=0.6.8 to documentation dependencies

Documentation:
- Add doc/draw/tree/plot_tree-iplotx.rst showcasing the iplotx integration

iosonofabio commented:
Thank you for the quick reply!

I cd into doc then ran the command above and get:

```bash
Traceback (most recent call last):
  File "/home/fabio/university/PI/projects/cogent3/doc/./doctest_rsts.py", line 16, in <module>
    from cogent3.app.composable import define_app
ModuleNotFoundError: No module named 'cogent3'
```

So I guessed it needs the `uv` extra-doc environment for this to run. I synced that and ran:

```bash
../.venv/bin/python doctest_rsts.py -f draw/plot_tree-iplotx.rst
```
which executed without errors in a couple seconds. I guess that means it can at least run through it without failures?

On a separate note, I'm rerunning the uv sphinx build process and it restarts from scratch which takes forever. Is that how it's supposed to be, no cache between builds?

One specific example, `app_tutorials/evo-natsel_zhang`, seems to hog all RAM and take forever... would be nice to be able to build a single file perhaps, would you know what command could achieve that?

Thank you!

GavinHuttley commented:
docbuilding is slooow, because we are doing some time consuming numerical optimisations (that example you gave is one such).

In terms of no cache between builds, sphinx seems to wipe everything if a single file has changed. Also, I use
`make html` as my command, with the virtual env active.

I've just asked ChatGPT and it says this is a common problem when using `nbsphinx`. It suggests adding the following lines to `conf.py`

```python
nbsphinx_execute = "auto"
nbexecution_mode = "cache"
```
It could be a hallucination, but it may work!

iosonofabio commented on doc/draw/tree/plot_tree-iplotx.rst:
```diff
@@ -0,0 +1,31 @@
+    :hide-code:
+
+    import set_working_directory
+    import matplotlib.pyplot as plt
```
Thanks, I moved it down and confirm it's not needed for the example, only to save the figure to file which is secret anyway.

iosonofabio commented on doc/draw/tree/plot_tree-iplotx.rst:
```diff
@@ -0,0 +1,31 @@
+
+Using iplotx with cogent3
+=========================
+
```
Great point, I've added a paragraph now.

iosonofabio commented:
Thank you, I've now added a paragraph of description and moved the secret code below (not needed for the actual plot).

I'm trying to run the docs, but it keeps failing at 72% with the following traceback about a missing file, plugin, or module (it's hard to tell ATM). Is this something you have encountered before by any chance?

(PS: The chatGPT suggestion does not seem to do anything... fair attempt though!)

```python
/home/fabio/university/PI/projects/cogent3/doc/app/app_overview/dstore.rst:12: WARNING: download file not readable: /home/fabio/university/PI/projects/cogent3/doc/app/app_overview/data/raw.zip [download.not_readable]
WARNING: Cell printed to stderr:
ERROR	Task(Task-2) stevedore.extension:extension.py:_load_plugins()- Could not load 'xmfa': module 'cogent3.parse.sequence' has no attribute 'XmfaParser'

WARNING: Cell printed to stderr:
/home/fabio/university/PI/projects/cogent3/src/cogent3/__init__.py:271: UserWarning: no moltype specified, defaulting to ASCII
  result = make_seq(


Extension error:
Cell raised uncaught exception:
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[23], line 3
      1 from cogent3 import load_aligned_seqs
----> 3 aln = load_aligned_seqs(
      4     "data/long_testseqs.fasta", moltype="dna", storage_backend="h5seqs_aligned"
      5 )
      6 aln

File ~/university/PI/projects/cogent3/src/cogent3/__init__.py:422, in load_aligned_seqs(filename, format_name, moltype, label_to_name, parser_kw, info, **kw)
    419 else:
    420     data = list(parser.loader(filename, **parser_kw))
--> 422 return make_aligned_seqs(
    423     data,
    424     label_to_name=label_to_name,
    425     moltype=moltype,
    426     source=filename,
    427     info=info,
    428     **kw,
    429 )

File /usr/lib/python3.13/functools.py:934, in singledispatch.<locals>.wrapper(*args, **kw)
    931 if not args:
    932     raise TypeError(f'{funcname} requires at least '
    933                     '1 positional argument')
--> 934 return dispatch(args[0].__class__)(*args, **kw)

File ~/university/PI/projects/cogent3/src/cogent3/core/alignment.py:7385, in make_aligned_seqs(data, moltype, label_to_name, info, source, annotation_db, offset, name_map, is_reversed, reversed_seqs, storage_backend, **kwargs)
   7382 if name_map is None:
   7383     name_map = make_name_map(data) or None
-> 7385 seqs_data = make_aligned_storage(
   7386     data,
   7387     moltype=moltype,
   7388     label_to_name=label_to_name,
   7389     offset=offset,
   7390     reversed_seqs=reversed_seqs,
   7391     storage_backend=storage_backend,
   7392     **kwargs,
   7393 )
   7394 # as they were handled in this function, we do not pass on:
   7395 # offset
   7396 # label_to_name
   7397 # reversed_seqs
   7398 # storage_backend
   7399 return make_aligned_seqs(
   7400     seqs_data,
   7401     moltype=moltype,
   (...)   7406     is_reversed=is_reversed,
   7407 )

File ~/university/PI/projects/cogent3/src/cogent3/core/alignment.py:7310, in make_aligned_storage(data, moltype, label_to_name, offset, reversed_seqs, storage_backend, **kwargs)
   7308 # plugin module is private only to exclude users, not developers
   7309 klass = cogent3._plugin.get_aligned_storage_driver(storage_backend)  # noqa: SLF001
-> 7310 return klass.from_seqs(**asd_kwargs)

File ~/university/PI/projects/cogent3/.venv/lib/python3.13/site-packages/cogent3_h5seqs/__init__.py:813, in AlignedSeqsData.from_seqs(cls, data, alphabet, **kwargs)
    811 path = kwargs.pop("storage_path", "memory")
    812 kwargs = {"mode": "w"} | kwargs
--> 813 return make_aligned(path, data=data, alphabet=alphabet, **kwargs)

File ~/university/PI/projects/cogent3/.venv/lib/python3.13/site-packages/cogent3_h5seqs/__init__.py:1308, in make_aligned(path, data, mode, in_memory, alphabet, offset, reversed_seqs, check, suffix, compression)
   1301     alphabet = c3_alphabet.make_alphabet(
   1302         chars=h5file.attrs["alphabet"],
   1303         gap=h5file.attrs["gap_char"],
   1304         missing=h5file.attrs["missing_char"],
   1305         moltype=mt,
   1306     )
   1307 check = h5file.mode == "r" if check is None else check
-> 1308 asd = AlignedSeqsData(
   1309     gapped_seqs=h5file,
   1310     check=check,
   1311     alphabet=alphabet,
   1312     offset=offset,
   1313     reversed_seqs=reversed_seqs,
   1314     compression=compression,
   1315 )
   1316 asd.filename_suffix = suffix
   1317 if data is not None:

TypeError: Can't instantiate abstract class AlignedSeqsData without an implementation for abstract methods 'get_pos_range', 'variable_positions'
```

GavinHuttley commented:
Do the following
```
pip install cogent3-h5seqs -U
```
also, merge in recent changes to the repo which fix that breakage.

iosonofabio commented:
Ok, now I polished the simple example and added a more complex one below to showcase some styling options.

I also adapted the `iplotx` codebase in a few ways to make the interaction more intuitive and smoother.

I also checked the docs build correctly locally after all this, which they do (with uv run sphinx...).

It seems to me like this would be ready to merge. Of course, happy to make more changes if requested.
